.github/workflows/plugin_test.yaml

```
@@ -0,0 +1,102 @@
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: test plugin
+
+on:
+  push:
+    branches: [generic_test_in_ci_cd]
+  pull_request:
+    branches: [master]
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python_version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install coverage coveralls
+      - name: Install package
+        run: |
+          pip install ".[dev]"
+      # Test for STM Plugin
+      - name: clone STM repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          repository: FAIRmat-NFDI/pynxtools-stm
+          path: pynxtools-stm
+      - name: Run STM tests
+        run: |
+          cd pynxtools-stm
+          git checkout TestFromPynxtools
+          git submodule sync --recursive
+          git submodule update --init --recursive --jobs=4
+          pip install .
+          pytest tests/test_reader.py::test_stm_reader
+      # # Test for MPES Plugin
+      # - name: clone MPES repo
+      #   uses: actions/checkout@v4
+      #   with:
+      #     fetch-depth: 0
+      #     repository: FAIRmat-NFDI/pynxtools-mpes
+      #     path: pynxtools-mpes
+      # - name: Run MPES tests
+      #   run: |
+      #     cd pynxtools-mpes
+      #     git checkout main
+      #     pip install .
+      #     pytest tests/.
+      # Test for XPS Plugin
+      - name: clone XPS repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          repository: FAIRmat-NFDI/pynxtools-xps
+          path: pynxtools-xps
+      - name: Run XPS tests
+        run: |
+          cd pynxtools-xps
+          git checkout main
+          pip install .
+          pytest tests/.
+      # # Test for EM Plugin
+      # - name: clone EM repo
+      #   uses: actions/checkout@v4
+      #   with:
+      #     fetch-depth: 0
+      #     repository: FAIRmat-NFDI/pynxtools-em
+      #     path: pynxtools-em
+      # - name: Run EM tests
+      #   run: |
+      #     cd pynxtools-em
+      #     git checkout main
+      #     pip install .
+      #     pytest tests/.
+      # # Test for APM Plugin
+      # - name: clone APM repo
+      #   uses: actions/checkout@v4
+      #   with:
+      #     fetch-depth: 0
+      #     repository: FAIRmat-NFDI/pynxtools-apm
+      #     path: pynxtools-apm
+      # - name: Run apm tests
+      #   run: |
+      #     cd pynxtools-apm
+      #     git checkout main
+      #     pip install .
+      #     pytest tests/.
```

Please at the commented test for pynx-ellips to plugin_test.yaml:
```suggestion
      #     pytest tests/.
      # # Test for ellipsometry Plugin
      # - name: clone ellipsometry repo
      #   uses: actions/checkout@v4
      #   with:
      #     fetch-depth: 0
      #     repository: FAIRmat-NFDI/pynxtools-ellips
      #     path: pynxtools-ellips
      # - name: Run ellipsometry tests
      #   run: |
      #     cd pynxtools-ellips
      #     git checkout main
      #     pip install .
      #     pytest tests/.
```